dotnet/src/Agents/Abstractions/ChatHistoryKernelAgent.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// A <see cref="KernelAgent"/> specialization bound to a <see cref="ChatHistoryChannel"/>.
+/// </summary>
+public abstract class ChatHistoryKernelAgent : KernelAgent, IChatHistoryHandler
+{
+    /// <inheritdoc/>
+    protected internal sealed override IEnumerable<string> GetChannelKeys()
+    {
+        yield return typeof(ChatHistoryChannel).FullName;
```

Comment:
Will the channel key always come from the channel instance?

If not, what is the use case for the channel not knowing it's own key?

Have you considered making the channel key a property of the `AgentChannel` class?